HPCC-25484 Move daliadmin helper functions for ESP WsDali

wangkx opened the pull request:
Rename daliadmin.cpp to daadmin.cpp. Create a new daliadmin.cpp.
Move the main(), usage(), and related code from daadmin.cpp back
to new daliadmin.cpp. Add the code to create a new library
daliadminlib which contains the helper functions in daadmin.cpp.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-14700, label: h25484 starts now.
The reason of this test is: Forced to re-test.
Commit ID: c2ec2cb0e106261b5d0261a6110a9ddf0aef92ff
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-218.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-218.ca-central-1.compute.internal
Sha: c2ec2cb0e106261b5d0261a6110a9ddf0aef92ff
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |146 |146 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |24 sec  (00:00:24) |
|setup (thor) |9 |9 |0 |39 sec  (00:00:39) |
|setup (roxie) |19 |19 |0 |17 sec  (00:00:17) |
|test (hthor) |959 |959 |0 |789 sec  (00:13:09) |
|test (thor) |870 |870 |0 |1079 sec  (00:17:59) |
|test (roxie) |1032 |1032 |0 |899 sec  (00:14:59) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  602 sec (00:10:02) |  109 sec (00:01:49) |   21 sec (00:00:21) |   20 sec (00:00:20) | 1567 sec (00:26:07) |   19 sec (00:00:19) | 2346 sec (00:39:06) |

jakesmith commented on dali/daliadmin/daadmin.hpp:
```diff
@@ -0,0 +1,88 @@
+
+static unsigned daliConnectTimeoutMs = 5000;
+
+extern da_decl void xmlSize(const char *filename, double pc);
```
da_decl doesn't look right here.. that's the decl. spec. from the dalibase library.
You _could_ move this functionality into that library, it's already linked by daliadmin and esp services.
But as not needed by most, it would be better to add it to a new library.

wangkx replied:
Will do

jakesmith commented on dali/daliadmin/daliadmin.cpp:
```diff
@@ -15,54 +15,12 @@
     limitations under the License.
 ############################################################################## */
 
-#include <unordered_map>
```
minor: but can you redo PR so that daliadmin.cpp is renamed to daadmin.cpp 1st?
That way, since most of the content is moving, the git history for all that content will be preserved.
Then daliadmin.cpp will be seen as new, but have much less content.

wangkx replied:
Will do

jakesmith commented on dali/daliadmin/daadmin.hpp:
```diff
@@ -0,0 +1,88 @@
+############################################################################## */
+
+#include "dautils.hpp"
+#include "workunit.hpp"
```
does this need to be in daadmin.hpp?

wangkx replied:
Not needed anymore. Will remove.

jakesmith reviewed (changes requested):
@wangkx - please see comments

wangkx committed:
HPCC-25484 Move daliadmin helper functions for ESP WsDali

Rename daliadmin.cpp to daadmin.cpp. Create a new daliadmin.cpp.
Move the main(), usage(), and related code from daadmin.cpp back
to new daliadmin.cpp. Add the code to create a new library
daliadminlib which contains the helper functions in daadmin.cpp.

This is the 1st commit. This commit does not have new daliadmin.cpp
which will be in the 2nd commit.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx commented on dali/daliadmin/daadmin.hpp:
```diff
@@ -0,0 +1,88 @@
+#include "dautils.hpp"
+#include "workunit.hpp"
+
+static unsigned daliConnectTimeoutMs = 5000;
```
Ok, I can move it to daadmin.cpp and daliadmin.cpp.

wangkx committed:
HPCC-25484 Add new daliadmin.cpp

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx commented on dali/daliadmin/daliadmin.cpp:
```diff
@@ -15,54 +15,12 @@
-#endif
+#include "daadmin.hpp"
 
 #define DEFAULT_DALICONNECT_TIMEOUT 5 // seconds
```
It is used by: daliConnectTimeoutMs = 1000 * props->getPropInt("timeout", DEFAULT_DALICONNECT_TIMEOUT);

HPCCSmoketest commented:
Process of PR-14700, label: h25484 starts now.
The reason of this test is: Base branch changed.
Commit ID: e1164e8d6e50e0e2e20b9750070f775fd1ddd653
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-183.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-183.ca-central-1.compute.internal
Sha: e1164e8d6e50e0e2e20b9750070f775fd1ddd653
Build: success
Milestone:Install hpccsystems-platform-community_8.0.1-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |150 |150 |0 |0 |0 |57 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |41 sec  (00:00:41) |
|setup (roxie) |19 |19 |0 |21 sec  (00:00:21) |
|test (hthor) |959 |959 |0 |782 sec  (00:13:02) |
|test (thor) |871 |871 |0 |1075 sec  (00:17:55) |
|test (roxie) |1032 |1032 |0 |895 sec  (00:14:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  596 sec (00:09:56) |  109 sec (00:01:49) |   21 sec (00:00:21) |   18 sec (00:00:18) | 1393 sec (00:23:13) |   19 sec (00:00:19) | 2164 sec (00:36:04) |

jakesmith reviewed (approved):
@wangkx - looks good

wangkx commented:
@richardkchapman this PR has 2 commits for better tracking the changing history. If you want me to squash them, please let me know. Otherwise, it is ready to be merged.
